File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/Account.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type AccountCoverage = BackboneElement & {
    coverage: Reference;
    priority?: number;
};

export type AccountGuarantor = BackboneElement & {
    party: Reference;
    onHold?: boolean;
    period?: Period;
};

export type Account = DomainResource & {
    description?: string;
    name?: string;
    servicePeriod?: Period;
    coverage?: AccountCoverage[];
    type?: CodeableConcept;
    guarantor?: AccountGuarantor[];
    status: string;
    identifier?: Identifier[];
    partOf?: Reference;
    subject?: Reference[];
    owner?: Reference;
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Account.py
from typing import Optional, List
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class Account_Coverage(BackboneElement):
    coverage: Reference
    priority: Optional[int] = None

class Account_Guarantor(BackboneElement):
    party: Reference
    on_hold: Optional[bool] = None
    period: Optional[Period] = None

class Account(DomainResource):
    description: Optional[str] = None
    name: Optional[str] = None
    service_period: Optional[Period] = None
    coverage: Optional[List[Account_Coverage]] = None
    type: Optional[CodeableConcept] = None
    guarantor: Optional[List[Account_Guarantor]] = None
    status: str
    identifier: Optional[List[Identifier]] = None
    part_of: Optional[Reference] = None
    subject: Optional[List[Reference]] = None
    owner: Optional[Reference] = None